### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by kingqabil**
- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical questions

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -3,4 +3,10 @@
 INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(1,'Agumon','2020-2-3',0,TRUE,10.23);
 INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(2,'Gabumon','2018-11-15',2,TRUE,8);
 INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(3,'Pikachu','2021-1-7',1,FALSE,15.04);
-INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(4,'Devimon','2017-5-12',5,TRUE,11);
\ No newline at end of file
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(4,'Devimon','2017-5-12',5,TRUE,11);
```
Comment (DammyShittu) [inline data.sql]: [OPTIONAL]
- I noticed that you're hardcoding values into the `id` column. What do you think about not having to bother about what number is next to be inserted into the `id` column?(imagine you had ten million rows) 😉
You can add the `GENERATED ALWAYS AS IDENTITY` constraint to the `id` column. This ensures that sequential integers that cannot be altered are generated each time a record is being created. I hope it makes sense. 😄 
You can go through this [article](https://www.sqltutorial.org/sql-identity/#:~:text=The%20GENERATED%20ALWAYS%20generates%20sequential,integers%20for%20the%20identity%20column.).

- You can also consider adding spaces after the commas `,` to enhance readability. 😉

File: data.sql [hunk 2]
```diff
@@ -3,4 +3,10 @@
...
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(5,'Charmander','2020-2-8',0,FALSE,-11);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(6,'Plantmon','2022-11-15',2,TRUE,-5.7);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(7,'Squirtle','1993-4-2',3,TRUE,-12.13);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(8,'Angemon','2005-5-12',1,TRUE,-45);
+INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES(9,'Boarmon','2005-5-7',7,TRUE,20.4);
```
Comment (DammyShittu) [inline data.sql]: - Kindly check the `date_of_birth` of `Angemon` and `Boarmon`. They were both born in `June` and not in `May`. 😃

File: queries.sql [hunk 3]
```diff
@@ -23,3 +24,67 @@ SELECT * FROM animals WHERE name!='Gabumon';
...
+ROLLBACK;
+
+-- Transactions to all animals with names ending with mon
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+--Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+
+BEGIN;
```
Comment (DammyShittu) [inline queries.sql]: - Kindly remove the `BEGIN` statement here as the transaction already began on line 40. Here 👇  is what the requirement says

File: queries.sql [hunk 4]
```diff
@@ -23,3 +24,67 @@ SELECT * FROM animals WHERE name!='Gabumon';
...
+-- What is the average weight of animals?
+SELECT AVG(weight_kg)
+FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT
+neutered,MAX(escape_attempts)
+FROM animals
+GROUP BY neutered
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg),MAX(weight_kg) FROM animals;
```
Comment (DammyShittu) [inline queries.sql]: - Awesome query!! 😀. However, the result of the query should show the `minimum` and `maximum` weight of each type(in this case, `species`) of animals. 

Here 👇 is an example of what the query could look like. 😉 

```
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```

File: queries.sql [hunk 5]
```diff
@@ -23,3 +24,67 @@ SELECT * FROM animals WHERE name!='Gabumon';
...
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts)
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
```
Comment (DammyShittu) [inline queries.sql]: - Awesome query!! 😀. However, the result of the query should show the average number of escape attempts per animal type(in this case, `species`). 

Here 👇 is an example of what the query could look like. 😉 

```
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @kingqabil, 👋 

Superb work done so far. 👏

Kudos to you. 👍

There are some issues that you still need to work on to go to the next milestone but, trust me, you are ALMOST THERE!

# Required Changes ♻️

_Please check the comments under the review._

# Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @kingqabil 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes
## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**